vm/src/compiler.rs

```diff
@@ -118,11 +121,23 @@ impl FunctionEnv {
         if let Slide(0) = i {
             return;
         }
-        debug!("{:?} {} {}", i, self.stack_size, i.adjust());
-        self.stack_size = (self.stack_size as i32 + i.adjust()) as VmIndex;
+        let adjustment = i.adjust();
+        debug!("{:?} {} {}", i, self.stack_size, adjustment);
+        if adjustment > 0 {
+            self.increase_stack(adjustment as VmIndex);
+        } else {
+            self.stack_size -= -adjustment as VmIndex;
+        }
         self.function.instructions.push(i);
```

Comment:
Add some spacing around some of these things to make it less dense. Also change `i` to `instruction`.